sycl/unittests/pi/TestGetPlugin.hpp

```diff
@@ -0,0 +1,44 @@
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+
+#include "BackendString.hpp"
+#include <algorithm>
+#include <detail/plugin.hpp>
+#include <functional>
+
+namespace pi {
+inline cl::sycl::detail::plugin initializeAndGet(cl::sycl::backend backend) {
+  auto plugins = cl::sycl::detail::pi::initialize();
+  auto it = std::find_if(plugins.begin(), plugins.end(),
+                         [=](cl::sycl::detail::plugin p) -> bool {
+                           return p.getBackend() == backend;
+                         });
+  if (it == plugins.end()) {
+    std::string msg = GetBackendString(backend);
+    msg += " PI plugin not found!";
+    throw std::runtime_error(msg);
+  }
+  return *it;
+}
+
+inline std::vector<cl::sycl::detail::plugin> initializeAndRemoveInvalid() {
```

Comment:
@bader @smaslov-intel 

I took on your feeback from https://github.com/intel/llvm/pull/1647, and came up with this. In the version of googletest that ships with LLVM, there is no way to skip tests at runtime (googletest would later introduce `GTEST_SKIP`). 

Instead I have written a wrapper around pi::initiaize for tests that drops plugins that will not behave well during unit testing. If the plugin cannot return any platforms or it returns an error code, it is dropped from the test. I plan on adding a log for this as well, so that this can be seen during testing as a warning.